larpandoracontent/LArObjects/LArDiscreteProbabilityVector.h

```diff
@@ -0,0 +1,393 @@
+/**
+ *  @file   larpandoracontent/LArObjects/LArDiscreteProbabilityVector.h
+ *
+ *  @brief  Header file for the lar discrete probability vector class
+ *
+ *  $Log: $
+ */
+#ifndef LAR_DISCRETE_PROBABILITY_VECTOR_H
+#define LAR_DISCRETE_PROBABILITY_VECTOR_H 1
+
+#include "Pandora/StatusCodes.h"
+
+#include <limits>
+#include <random>
+#include <utility>
+#include <vector>
+
+namespace lar_content
+{
+
+/**
+ *  @brief  DiscreteProbabilityVector class
+ */
+class DiscreteProbabilityVector
+{
+public:
+
+    template <typename TX, typename TY>
+    using InputDatum = std::pair<TX, TY>;
+
+    template <typename TX, typename TY>
+    using InputData = std::vector<InputDatum<TX, TY> >;
+
+    typedef std::vector<float> ResamplingPoints;
+
+    /**
+     *  @brief  Constructor
+     *
+     *  @param  inputData the data used to construt the probability vector
+     *  @param  xUpperBound the upper bound of the probability vector
+     *  @param  useWidths bool controlling whether the bin widths are used in calculations
+     */
+    template <typename TX, typename TY>
+    DiscreteProbabilityVector(const InputData<TX, TY> &inputData, const TX xUpperBound, const bool useWidths);
```

Comment:
I think that, based on the code's current structure, it can only handle trivial objects like int/float.  A bit of work would be needed for more complex cases so it's probably fine to leave this one as is.